### PR TITLE
Asset browser ui

### PR DESCRIPTION
**Pull request by AlexanderOtavka**
Add dynamic and collision shape selection to add to world flow from asset browser.

### Testing
1. Open asset browser
2. Select an fbx file and click add to world
3. A dialog identical to the add model dialog should appear
4. Try different configurations and different models
5. Shape type, dynamic state, and model URL should all be reflected in the created entities
6. When you add to world an object that has dynamic true, the object should have a gravity of -10.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: c25b87c33ff213a915b73eceba30812ed4c6ea17 available at: https://gist.github.com/925dd70fab4904193be1f510138f5828<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8301-c25b87c.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8301-c25b87c.dmg

**Comment by RyanDowne** [comment]
This is looking good but I am not seeing the -10 gravity when dynamic is selected.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 344ec25bcd3ede709e5edc0553e01769bad75f32 available at: https://gist.github.com/925dd70fab4904193be1f510138f5828<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8301-344ec25.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8301-344ec25.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 90154158b6eb1a5099fae6ece1b0b3cd5dbbc2f6 available at: https://gist.github.com/925dd70fab4904193be1f510138f5828<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8301-9015415.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8301-9015415.dmg

**Comment by Nex-Pro** [comment]
**PR Build:** #8211
**Testing Architecture:** Desktop PC
**CPU:** 4.4 ghz Intel i7
**RAM:** 16 GB
**GPU:** Nvidia gtx770 2gb
**Testing OS:** win 10 pro
**HMD:** None
**Hand Controllers:** None

### QA PASS

Works fine.  Models with 'dynamic' set acquire a gravity of -10 Y.

Selecting different hull types results in the appropriate collisions. For example the firepit when simple has a smooth top. Otherwise avatars collide with the rocks in the fire.

***Only issue seen was some objects not acting dynamically unless collided with. But I don't think that relates to this PR.***

File: libraries/entities/src/EntityScriptingInterface.h [hunk 1]
```diff
@@ -86,7 +86,8 @@ public slots:
     Q_INVOKABLE QUuid addEntity(const EntityItemProperties& properties, bool clientOnly = false);
 
     /// temporary method until addEntity can be used from QJSEngine
-    Q_INVOKABLE QUuid addModelEntity(const QString& name, const QString& modelUrl, const glm::vec3& position);
+    Q_INVOKABLE QUuid addModelEntity(const QString& name, const QString& modelUrl, const QString& shapeType, bool dynamic,
+                                     const glm::vec3& position, const glm::vec3& gravity);
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityScriptingInterface.h]: Normally I would object to the `EntityScriptingInterface::addModelEntity()` method: it is not general enough and is therefore less useful as general `EntityScriptingInterface` API -- what happens when I want to add a `ModelEntityItem` with a specific: rotation, velocity, dimensions, or any of a dozen other properties?

However, I note that the comment above says that this is a "temporary method until addEntity can be used from QJSEngine", and that it is only ever called in one spot so far.  If we want to continue to punt this technical debt down the road then it doesn't block this PR.  I'll ask around to see what others want regarding this issue.

**Comment by AndrewMeadows** [comment]
The problem where dynamic objects are "inactive" until some other dynamic object moves near them (thereby waking them up in the physics simulation) is fundamental to how we load objects in the world and is not new to this PR.

File: libraries/entities/src/EntityScriptingInterface.h (hunk 1 above)
Comment (AndrewMeadows) [inline libraries/entities/src/EntityScriptingInterface.h]: We're going to kick this forward, but hopefully for only a week or two.  Zander will try to use a QVariant.

**Pull request merged**